src/qmapshack/gis/trk/CGisItemTrk.cpp

```diff
@@ -3402,3 +3410,32 @@ QMap<searchProperty_e, CGisItemTrk::fSearch> CGisItemTrk::initKeywordLambdaMap()
 
     return map;
 }
+
+void CGisItemTrk::toRoute()
+{
+    qint32 idx1, idx2;
+    QString routeName;
+
+    getMouseRange(idx1, idx2, true);
+
+    if(NOIDX == idx1)
+    {
+        routeName = getName();
+    }
+    else
+    {
+        routeName = getName() + QString(" (%1-%2)").arg(idx1).arg(idx2);
+    }
+
+    CTrkToRteDialog dlg(getParentProject(), routeName);
```

Comment:
Be careful! getParentProject() does a dynamic cast and might return with a nullptr.

Reply:
Could this really return null? Since I convert from an existing track, there must exist also a parent project where the track is assigned to.

Reply:
Is for getParentProject really a dynamic cast required? As far as i can see, it is always a downcast to IGisProject, so should be a static cast enough?

Reply:
Better safe than sorry. Never trust a getter method returning a pointer.

parent() returns a QTreeWidgetItem. That can be everything and must not be necessarily a IGisProject item. Also within the code GIS items are created without a parent (but that probably never applies here).

Anyways it's good practice to check and no it's not done everywhere because it's often forgotten (including code written by me).

Reply:
I have added a check.